src/hotspot/cpu/x86/templateInterpreterGenerator_x86_64.cpp

```diff
@@ -445,3 +445,20 @@ address TemplateInterpreterGenerator::generate_math_entry(AbstractInterpreter::M
   return entry_point;
 }
 
+address TemplateInterpreterGenerator::generate_currentThread() {
+
+  address entry_point = __ pc();
+
+  __ movptr(rscratch2, Address(r15_thread, JavaThread::threadObj_offset()));
+
+  // Only IN_HEAP loads require a thread_tmp register
+  __ access_load_at(T_OBJECT, IN_NATIVE, rax,
+                    Address(rscratch2, 0), rscratch1, noreg);
```

Comment:
I don't like the hardcoded 0 here. You should use MacroAssembler::resolve_oop_handle() instead. (That has the 0 too, I believe this should be changed to a constant like OopHandle::obj_offset_in_bytes() or so, but that is another story.)

Reply:
Yes, thanks, agreed, fixed.